code/PostgreSQLSchemaManager.php

```diff
@@ -494,9 +494,15 @@ private function alterTableAlterColumn($tableName, $colName, $colSpec)
         // First, we split the column specifications into parts
         // TODO: this returns an empty array for the following string: int(11) not null auto_increment
         //         on second thoughts, why is an auto_increment field being passed through?
-
-        $pattern = '/^([\w(\,)]+)\s?((?:not\s)?null)?\s?(default\s[\w\.\']+)?\s?(check\s[\w()\'",\s]+)?$/i';
+        $pattern = '/^([\w(\,)]+)\s?((?:not\s)?null)?\s?(default\s[\w\.\'\\\\]+)?\s?(check\s[\w()\'",\s\\\\]+)?$/i';
```

Comment:
This regex is meant to split up $colSpec and identify the various parts ofthe column definition. e.g.:
```
varchar(255) default 'Internal' check ("RedirectionType" in ('Internal', 'External', null))
```

This works OK until your enum starts having backslashes, as our ClassName now have.
```
varchar(255) default 'SS\Test\Polymorphic\Player' check ("ClassName" in ('SS\Test\Polymorphic\Player', 'Player', null))
```

All I'm changing here, is allowing the defaults and the list of accepted values to contain backslashes.

Reply:
Let's consider other crazy characters in here (apart from backslashes)

Reply:
potentially could try to accomodate this regex - `(?:(?:"(?:\\"|[^"])+")|(?:'(?:\\'|[^'])+'));` (found it on Stack - https://stackoverflow.com/questions/1017051/php-to-extract-a-string-from-double-quote)
I guess if we exclude the quotes it might do the trick. I'll play with it.

Reply:
I'll have a go at making this a bit smarter.

Reply:
I tried throwing special characters at our enum implementation in MySQL and PostgreSQL. I tried `~!@#$%^&*(-+?<\/'"`

The only one that got different results was `'`. PostgreSQL failed on dev/build while MySQL was happy. That was for a brand new column, not one that needed updating. It's probably worth logging as a known issue, but I'm not sure it's worth fixing through this PR.

Reply:
sounds good to me